definitions/artifacts/azure-databricks-workspace.json

```
@@ -0,0 +1,104 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema",
+	"$id": "https://schemas.massdriver.cloud/definitions/artifacts/azure-databricks-workspace.json",
+	"$md": {
+		"access": "public"
+	},
+	"type": "object",
+	"title": "Azure Databricks Workspace",
+	"description": "",
+	"additionalProperties": false,
+	"required": [
+		"data",
+		"specs"
+	],
+	"properties": {
+		"data": {
+			"title": "Artifact Data",
+			"type": "object",
+			"required": [
+				"infrastructure",
+				"security"
+			],
+			"properties": {
+				"infrastructure": {
+					"title": "Infrastructure configuration",
+					"type": "object",
+					"required": [
+						"ari",
+						"url",
+						"raw_container_ari",
+						"scratch_container_ari",
+						"enriched_container_ari"
+					],
+					"properties": {
+						"ari": {
+							"$ref": "../types/azure-resource-id.json"
+						},
+						"url": {
+							"title": "Azure Databricks workspace URL",
+							"$ref": "../types/https-url.json"
+						},
+						"raw_container_ari": {
+							"title": "Raw Container",
+							"description": "The name of the container where the raw data will be stored.",
+							"$ref": "../types/azure-infrastructure-storage-container.json"
+						},
+						"scratch_container_ari": {
+							"title": "Scratch Container",
+							"description": "The name of the container where intermediate data transformations or experimental analyst data will be stored.",
+							"$ref": "../types/azure-infrastructure-storage-container.json"
+						},
+						"enriched_container_ari": {
+							"title": "Enriched Container",
+							"description": "The name of the container where curated data assets will be stored.",
+							"$ref": "../types/azure-infrastructure-storage-container.json"
+						}
+					}
+				},
+				"security": {
+					"title": "Security Configuration",
+					"type": "object",
+					"required": [
+						"raw-read-write",
+						"raw-read-only",
+						"scratch-read-write",
+						"enriched-read-write",
+						"enriched-read-only"
+					],
+					"properties": {
+						"iam": {
+							"title": "IAM Configuration",
+							"properties": {
+								"raw-read-write": {
+									"$ref": "../types/azure-security-iam.json"
+								},
+								"raw-read-only": {
+									"$ref": "../types/azure-security-iam.json"
+								},
+								"scratch-read-write": {
+									"$ref": "../types/azure-security-iam.json"
+								},
+								"enriched-read-write": {
+									"$ref": "../types/azure-security-iam.json"
+								},
+								"enriched-read-only": {
+									"$ref": "../types/azure-security-iam.json"
+								}
+							}
+						}
+					}
```

Pretty sure this still needs to change. Check my work on the curly braces I removed, but I believe this is all you need here. 

```suggestion
					"properties": {
						"iam": {
                                                  "$ref": "../types/azure-security-iam.json"
						}
					}
```

---

oh I see that type is a pattern properties map! good catch.

---

Your Terraform will look more like what you wrote here

```terraform
security = {
  iam = {
    # we've talked about using slashes instead of dashes for these keys
    # let us know what you think
    "read/write": {
      "role": "Name of the role",
      "scope": "subscriptions/etc/etc/etc"
    }
    "encriched-read/write": {
      "role": "Name of the role",
      "scope": "subscriptions/etc/etc/etc"
    }
    "enriched-read": {
      "role": "Name of the role",
      "scope": "subscriptions/etc/etc/etc"
    }
    .....
  }
}
```

---

Aye aye sir 👍

---

@WillBeebe is there a reason that type's patternProperties regex doesn't allow dashes? it'd be helpful to have dashes or colon in the keys so I could have `raw-read/write` or `enriched:readOnly`. I could add it in this PR but wanted to ask if it was enforcing a constraint for a valid reason.

---

Great question actually! I think the intention is that it supports dashes and slashes. This might be a thing @chrisghill and I talked about but never made it into code. The AWS and GCP ones only support slashes and a-z as well! I'm pretty sure we do want to add the dash to that pattern property, I'll let Chris chime in to confirm.

---

captured a naming convention / style in the guidelines https://massdriver.slab.com/posts/artifact-authoring-guidelines-c4sppru2#hv2ip-iam-security-blocks-naming-convention

---

For AWS and GCP:
https://github.com/massdriver-cloud/artifact-definitions/pull/69